Design.md

```diff
@@ -227,8 +229,26 @@ This API will enable or disable user on Vault.
 
 
 ### Create S3 Credential
-This API creates S3 credential for the user.
-1. `generate-access-key` api will be called using assumed role credentials.
+This API creates S3 credentials for the user.
+1. The Generate Access Key API will be called using assumed role credentials.
+1. Encrypt the secret key (For more, see [SecretKey-Encryption-Strategy](#SecretKey-Encryption-Strategy)). Store the encrypted secret key on Redis Sentinel in the hash named "osis:s3credentials", with the hash key formatted as `<Username>__<AccessKeyID>`.
+1. Log the access key ID _(Log location and tools for logging are yet to be decided)_
+
+**Important Notes:**
+* Redis Sentinel is subject to an ongoing rolling deployment.
```

Comment:
@saitharunthodupunuru @terrydervaux 
What is the difference between "an ongoing rolling upgrade" and "a rolling upgrade"? This seems superfluous the way it is written. Was our intention to insinuate that the OSIS data would not survive if Redis Sentinel was upgraded in parallel?

Reply:
@scality-trevorbenson 
 Yes the idea here is to do rolling upgrades to Redis sentinel to avoid any loss of OSIS data during the parallel upgrade. Please suggest if you think otherwise.
 
 Here "an ongoing rolling upgrade" was added to mention that any ongoing upgrades should be a rolling deployment. And "a rolling upgrade" may suggest it to be generic. I felt "an ongoing rolling upgrade" has no ambiguity.

Reply:
@saitharunthodupunuru: I suggest you remove that line because it creates confusion. Instead you can change the next line to say something like:
> If the Redis Sentinel data loss happens due to, for instance, an entire cluster failure, OSIS will not be able to list access keys created before the Redis Sentinel data loss. 
> In that case, The vCloud Director's OSE service must be restarted with the following command: